pandas/io/parsers/readers.py

```diff
@@ -511,8 +511,14 @@ def _read(
 ):
     """Generic reader of line files."""
     if kwds.get("date_parser", None) is not None:
```

Comment:
This is hard to read. Please do it somehow like this:

```
if kwds.get("date_parser", None) is not None and kwds.get("pars_dates", None) is None:
    kwds.get["pars_dates"] = True
elif kwds.get("pars_dates", None) is None:
    kwds.get["pars_dates"] = False
```

Reply:
Correct! I have made the changes so now it is more readable, added a test to see if the parser is not called when we set `parse_dates` to `False`.
Some tests are failing for an obscure reason as they are not related to the code change (and they are also failing on other builds)